packages/babel-parser/src/parser/expression.js

```diff
@@ -560,10 +560,15 @@ export default class ExpressionParser extends LValParser {
       stop: false,
     };
     do {
+      const oldMaybeInAsyncArrowHead = this.state.maybeInAsyncArrowHead;
+      if (state.maybeAsyncArrow) {
+        this.state.maybeInAsyncArrowHead = state.maybeAsyncArrow;
+      }
```

Comment:
Nit: inside the `if` `state.maybeAsyncArrow` is always `true`.

```suggestion
      if (state.maybeAsyncArrow) this.state.maybeInAsyncArrowHead = true;
```

Reply:
@nicolo-ribaudo Ah, yeah. Just updated the PR.